src/x11_utils.rs

```diff
@@ -37,6 +37,47 @@ pub trait TryParse: Sized {
     fn try_parse(value: &[u8]) -> Result<(Self, &[u8]), ParseError>;
 }
 
+/// A representation of the header of a request.
+#[derive(Debug, Clone, Copy)]
+pub struct RequestHeader {
+    /// The major opcode of the request.
+    pub major_opcode: u8,
+    /// The minor opcode of the request (which, for some requests, may not be an
+    /// opcode at all).
+    pub minor_opcode: u8,
+    /// The remaining length of the request, in units of 4 bytes. Unlike the wire format,
+    /// this does *not* include the header itself. If the BigRequests extension is enabled
+    /// this can be greater than u16::max_value - 1.
```

Comment:
If I understand this correctly: Could you make it explicit that the length of the header is 1 (or 2 if big requests is used) "4 byte units".

However, is this field actually needed? All length checks should "follow naturally" from `TryParse` implementations calling `.get` on the request slice.

Reply:
Comment addressed. This data is ultimately still needed by whoever is driving the parser to make sure they've waited for enough bytes from the socket/etc before attempting to parse it.